src/d/actor/d_a_tornado.cpp

```diff
@@ -4,72 +4,342 @@
+static BOOL daTornado_Draw(daTornado_c* i_this) {
+    return i_this->draw();
 }
 
 /* 00000484-00000C4C       .text execute__11daTornado_cFv */
 BOOL daTornado_c::execute() {
-    /* Nonmatching */
+    static cXyz wind_scale(20.0f,20.0f,20.0f);
+
+    float fVar8;
+
+    mBck.play();
+    mBtkFrame += 1.0f;
+    if (mBtkFrame >= mBtk.getBtkAnm()->getFrameMax()) {
+        mBtkFrame -= mBtk.getBtkAnm()->getFrameMax();
+    }
+
+    mBtkUnderFrame += 1.0f;
+    if (mBtkUnderFrame >= mBtkUnder.getBtkAnm()->getFrameMax()) {
+        mBtkUnderFrame -= mBtkUnder.getBtkAnm()->getFrameMax();
+    }
+    mAngle2 += 250;
+    mAngle1 += 500;
+    m32c = 10000.0f;
+    u32 param = fopAcM_GetParam(this);
+    if (param == 1) {
+        if (dComIfGp_getShipActor() != NULL) {
+            fVar8 = (dComIfGp_getShipActor()->current.pos.y - home.pos.y) / 500.0f;
+            if (fVar8 < 0.0f) {
+                fVar8 = 0.0f;
+            } else {
+                if (fVar8 > 1.0f) {
+                    fVar8 = 1.0f;
+                }
+            }
+        } else {
+            fVar8 = 1.0f;
+        }
+    } else if (param == 0 && dComIfGp_getShipActor() != NULL && dComIfGp_getShipActor()->getTornadoActor() != NULL) {
+        fVar8 = 0.25f;
+    } else {
+        fVar8 = 1.0f;
+    }
+
+    for (s32 i = 0; i < 11; i++) {
+        float sin = cM_ssin(mAngle1 - 0x1000 * i);
+        float tmp1 = fVar8 * joint_offset[i];
+        float fVar1 = tmp1 * (sin + 1.0f);
+        short angle2 = mAngle2 - 0x1800 * i;
+        mJointX[i] = cM_ssin(angle2) * fVar1 * scale.x;
+        mJointZ[i] = cM_scos(angle2) * fVar1 * scale.x;
+    }
+
+    param = fopAcM_GetParam(this);
+    if (param == 0) {
+        if (!dComIfGp_event_runCheck()) {
+            short target = fopAcM_searchActorAngleY(this, dComIfGp_getLinkPlayer());
+            cLib_addCalcAngleS(&current.angle.y, target, 10, 0x1000, 0x100);
+            cLib_chaseF(&speedF, 20.0f, 0.2f);
+        }
+        if (dComIfGp_checkPlayerStatus0(0, daPyStts0_SHIP_RIDE_e) && dComIfGp_getShipActor() != NULL) {
+            daShip_c* ship = dComIfGp_getShipActor();
+            cXyz diff = ship->current.pos - current.pos;
+            if (diff.abs2XZ() < 100000000.0f) {
+                ship->onTornadoFlg(fopAcM_GetID(this));
+                speedF = 0.0f;
+            }
+        }
+
+        if (!dComIfGp_event_runCheck()) {
+            fopAcM_posMoveF(this, NULL);
+            cXyz diff = current.pos - home.pos;
+            diff.y = 0;
+            if (diff.abs2XZ() > 56250000) {
+                diff.normalize();
+                current.pos = home.pos + diff * 7500.0f;
+            }
+        }
+
+        if (mPtclTimer != 0) {
+            mPtclTimer -= 1;
+        } else {
+            mPtclTimer = 10;
+            dComIfGp_particle_set(0x8213, &current.pos, NULL, (cXyz*)&wind_scale);
+        }
+
+        fopAcM_seStartCurrent(this, JA_SE_OBJ_TORNADE_SUS, 100);
+
+        if (dComIfGs_isEventBit(10000)) {
```

Comment:
```suggestion
        if (dComIfGs_isEventBit(0x2710)) {
```
Memory bits should be hex (it's a coincidence that it looks like a round decimal number)